extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -279,6 +279,31 @@ export class AnalyticsRoot {
     });
   }
 
+  /**
+   * @param {!Element} context
+   * @param {string} selector DOM query selector.
+   * @return {!Promise<!Array<!Element>>} Element corresponding to the selector.
+   */
+  getElementsByQuerySelectorAll_(context, selector) {
+    // Wait for document-ready to avoid false missed searches
+    return this.ampdoc.whenReady().then(() => {
+      let foundElements;
+      try {
+        foundElements = this.getRoot().querySelectorAll(selector);
+      } catch (e) {
+        userAssert(false, `Invalid query selector ${selector}`);
+      }
+      const elements = [];
+      for (let i = 0; i < foundElements.length; i++) {
+        if (this.contains(foundElements[i])) {
```

Comment:
`this.contains`?

Reply:
[Used in the original getAmpElements()](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/analytics-root.js#L273-L277), to make sure that the elements selected are descendants of the `analytics-root`: [defined here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/analytics-root.js#L175)